sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/HttpRedirectOptions.java

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.generic.core.http.models;
+
+import com.generic.core.http.Response;
+import com.generic.core.http.policy.RequestRedirectCondition;
+import com.generic.core.models.HeaderName;
+import com.generic.core.util.ClientLogger;
+
+import java.util.EnumSet;
+import java.util.function.Predicate;
+
+/**
+ * Options to configure the redirect policy's behavior.
+ */
+public final class HttpRedirectOptions {
+    private static final ClientLogger LOGGER = new ClientLogger(HttpRedirectOptions.class);
+    private final int maxAttempts;
+    private final EnumSet<HttpMethod> allowedRedirectHttpMethods;
+    private final HeaderName locationHeader;
+    private Predicate<RequestRedirectCondition> shouldRedirectCondition;
+
+    /**
+     * Creates an instance of {@link HttpRedirectOptions} with values for {@code maxAttempts}
+     * defaulting to 3, if not specified, and the default set of allowed HTTP methods {@link HttpMethod#GET}
```

Comment:
I think the defaults should be set on the policy and these options shouldn't document what the policy is doing. This class should just allow all properties to be optional and the policy can determine what to do when they are not set.